### ISSUE TITLE
`jax.scipy.linalg.expm` produces nan values for very matrices with large negative values

### ISSUE THREAD

**Issue by zaccharieramzi** [google/jax#13985]
### Description

When I am using `jax.scipy.linalg.expm`, I sometimes want to compute the exponential of matrices whose values are very large in magnitude but negative. I am therefore expecting the output to be very close to the all null matrix.
However, I notice that I get all nans instead.

I think there is a trick implemented in scipy to make `expm` work even for very large negative values, but I am not sure which it is.
I see that there was a discussion [here](https://github.com/scipy/scipy/issues/11839). Not sure that the trick mentioned in [this comment](https://github.com/scipy/scipy/issues/11839#issue-597926417), i.e. the squaring/scaling trick is the one that actually got used to solve the issue, as I do not see this implemented in the subsequent [PR](https://github.com/scipy/scipy/pull/11840/files).

The code to reproduce would be the following:
```python
import jax
import jax.numpy as jnp
from jax.scipy.linalg import expm as jexpm
import numpy as np
from scipy.linalg import expm

dimension_x = 10
t = 1e6

# the scipy way
sqrt_H = np.random.normal(size=(dimension_x, dimension_x))
H = np.dot(sqrt_H, sqrt_H.T)
print(expm(-t*H))

# the jax way
sqrt_H = jax.random.normal(jax.random.PRNGKey(1), (dimension_x, dimension_x))
H = jnp.dot(sqrt_H, sqrt_H.T)
print(jexpm(-t*H))
```

### What jax/jaxlib version are you using?

jax==0.3.25 jaxlib==0.3.25

### Which accelerator(s) are you using?

CPU

### Additional system info

Python 3.9.13, Linux (Ubuntu 20.04)

**Comment by zaccharieramzi** [comment]
Actually since this is rather a feature request I will close this and open a new one with the correct issue label.
